### PR TITLE
Issue#SB-21003 feat: added discussion forum

### PR DESCRIPTION
**Pull request by swetha-aggidevara**

### CODE REVIEW

File: src/app/client/src/app/modules/discussion/components/forum/forum.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  initEditor() {
+    const userName = _.get(this.userService.userProfile, 'userName');
+    this.http.get(`/get/user/sessionId?userName=` + userName).subscribe((data: any) => {
+
+    jQuery('#discussionIframe').iziModal({
+      // title: '',
+      iframe: true,
+      iframeURL: `https://dev.sunbirded.org/discussions/auth/sunbird-oidc/callback${data.id}&returnTo=/category/15`,
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/discussion/components/forum/forum.component.ts:30]: Can you remove the host from here.

File: src/app/client/src/app/modules/discussion/components/forum/forum.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        });
+      }
+    }).iziModal('open');
+  });
+  }
+
+  public closeModal() {
+    // this.showLoader = true;
+    if (document.getElementById('discussionIframe')) {
+      document.getElementById('discussionIframe').remove();
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/discussion/components/forum/forum.component.ts:51]: Use Jquery itself to remove.

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html [hunk 3]
```diff
@@ -54,6 +54,14 @@ <h5 class="textbook__title sb-color-primary font-weight-bold mb-0">{{courseHiera
                 </svg></i>
                 {{resourceService?.frmelmnts?.lbl?.copy}}
               </button>
+              <button class="sb-btn sb-btn-normal sb-btn-link sb-btn-link-primary sb-left-icon-btn ml-8 sb-btn-icon-fix" 
+              appTelemetryInteract [telemetryInteractObject]="courseInteractObject"
+              [telemetryInteractEdata]="{id:'discussion' , type: 'click' , pageid:'course-consumption'}"
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html:59]: We have to CDATA with CourseId & object with course Id

File: src/app/routes/ssoRoutes.js [hunk 4]
```diff
@@ -423,6 +423,10 @@ module.exports = (app) => {
     }
   });
 
+  app.all('/get/user/sessionId', (req, res) => {
+    console.log('collectionEditorURL', req.query);
```
Comment (vinukumar-vs) [inline src/app/routes/ssoRoutes.js:427]: Remove this console message

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.ts [hunk 5]
```diff
@@ -263,6 +263,12 @@ export class CourseConsumptionHeaderComponent implements OnInit, AfterViewInit,
     }
   }
 
+
+  openDiscussionForum() {
+    this.router.navigate(['/discussions']);
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.ts:268]: What will happen when we click on browser back after navigating to discussion forum

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/discussion/components/forum/forum.component.ts (hunk 2 above)
Comment (swetha-aggidevara) [reply src/app/client/src/app/modules/discussion/components/forum/forum.component.ts:49]: If we only jQuery on click of the browser back button previous page is not loading

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html [hunk 6]
```diff
@@ -54,6 +54,15 @@ <h5 class="textbook__title sb-color-primary font-weight-bold mb-0">{{courseHiera
                 </svg></i>
                 {{resourceService?.frmelmnts?.lbl?.copy}}
               </button>
+              <button class="sb-btn sb-btn-normal sb-btn-link sb-btn-link-primary sb-left-icon-btn ml-8 sb-btn-icon-fix" 
+              appTelemetryInteract [telemetryInteractObject]="courseInteractObject"
+              [telemetryInteractEdata]="{id:'discussion' , type: 'click' , pageid:'course-consumption'}" 
+              [telemetryInteractCdata]="[{id: courseHierarchy?.identifier, type: courseHierarchy?.primaryCategory}]"
```
Comment (vinukumar-vs) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-header/course-consumption-header.component.html:60]: Dont use primaryCategory
courseHierarchy?.primaryCategory. Use static string

**Review by vinukumar-vs — APPROVED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**